CRISPResso2/CRISPRessoCORE.py

```diff
@@ -501,6 +862,488 @@ def custom_encoder(obj):
+                                " DEL="+("&".join([';'.join(x) for x in del_inds])) +\
+                                " INS="+("&".join([';'.join(x) for x in ins_inds])) +\
+                                " SUB=" + ("&".join([';'.join([str(y) for y in x]) for x in sub_inds])) +\
+                                " ALN_REF=" + ('&'.join([new_variant['variant_'+name]['aln_ref'] for name in new_variant['aln_ref_names']])) +\
+                                " ALN_SEQ=" + ('&'.join([new_variant['variant_'+name]['aln_seq'] for name in new_variant['aln_ref_names']]))
+                        new_variant['crispresso2_annotation'] = crispresso2_annotation
+                        fastq_write_out_handle.write(f"{fastq1_id.strip()}\n{new_variant['ref_aln_details'][0][1].replace('-', '')}\n+{crispresso2_annotation}\n{new_variant['ref_aln_details'][0][4]}\n")
+
+                    if new_variant['caching_is_ok']:
+                        variantCache[lookup_fastq_seq] = new_variant
+                    else:
+                        if len(new_variant['aln_ref_names']) == 1 or args.expand_ambiguous_alignments:                   
+                            for name in new_variant['aln_ref_names']:
+                                match_name = "variant_" + name
+                                if READ_LENGTH == 0:
+                                    READ_LENGTH = len(new_variant[match_name]['aln_seq'])
+                                N_GLOBAL_SUBS += (new_variant[match_name]['substitution_n'] + new_variant[match_name]['substitutions_outside_window'])
+                                N_SUBS_OUTSIDE_WINDOW += new_variant[match_name]['substitutions_outside_window']
+                                N_MODS_IN_WINDOW += new_variant[match_name]['mods_in_window']
+                                N_MODS_OUTSIDE_WINDOW += new_variant[match_name]['mods_outside_window']
+                                if new_variant[match_name]['irregular_ends']:
+                                    N_READS_IRREGULAR_ENDS += 1
+                        alignment_key = new_variant['variant_' + new_variant['aln_ref_names'][0]]['aln_seq']
+                        if alignment_key in variantCache:
+                            variantCache[alignment_key]['count'] += 1
+                        else:
+                            variantCache[alignment_key] = new_variant
+                else:
+                    N_COMPUTED_NOTALN += 1
+                    if args.fastq_output:
+                        crispresso2_annotation = " ALN=NA" +\
+                            " ALN_SCORES=" + ('&'.join([str(x) for x in new_variant['aln_scores']])) +\
+                            " ALN_DETAILS=" + ('&'.join([','.join([str(y) for y in x]) for x in new_variant['ref_aln_details']]))
+                        fastq_write_out_handle.write(f"{fastq1_id.strip()}\n{new_variant['ref_aln_details'][0][1].replace('-', '')}\n+{crispresso2_annotation}\n{new_variant['ref_aln_details'][0][4]}\n")
```

Comment:
This may make a separate `get_crispresso2_annotation` function (as mentioned above) difficult... happy to hear your thoughts either way.